source/fuzz/transformation_swap_function_variables.cpp

```diff
@@ -69,6 +69,11 @@ void TransformationSwapFunctionVariables::Apply(
       ir_context->get_def_use_mgr()->GetDef(message_.result_id2());
 
   std::swap(*instruction1, *instruction2);
+
```

Comment:
After thinking about it, I don't think we should swap the _contents_ of two instructions. That seems difficult in terms of trying to preserve the data structures in ir_context. E.g. the DecorationManager stores pointers to Instruction*, so we really don't want to be changing the pointees. 

Instead, try this:

```
std::unique_ptr<opt::Instruction> temp_instruction =
      MakeUnique<opt::Instruction>();

  temp_instruction->InsertBefore(instruction1);
  instruction1->InsertAfter(instruction2);
  instruction2->InsertAfter(temp_instruction.get());
  temp_instruction->RemoveFromList();
```

With this change, I think you can also remove the lines below that update the def use manager. I don't think we need to update or invalidate any analyses here.

One analysis that you might think you need to invalidate is the dominator analysis. However, the dominates relationship for instructions within a block is computed on-the-fly based on the instruction list, so nothing else needs to be updated or invalidated.

Reply:
Done, Thanks for the clarification @paulthomson. Can I understand what is the benefits of a known "dominates relationship for instructions"?

Reply:
This is explained in the SPIR-V introduction slides, which we will go through again in our meeting...now! :-)

Reply:
Thanks :-)